baselines/hfedxgboost/README.md

```diff
@@ -0,0 +1,317 @@
+```
+2- Install **poetry** at the system level: https://python-poetry.org/docs/ --> running curl -sSL https://install.python-poetry.org | python3 -(dont' forget to add to your .bashrc or equivalent)
+
+3- Then install a version of Python of your choice via pyenv, eg: `pyenv install 3.10.6`
+
+4- In the this directory (i.e. `/baselines/hfedxgboost`) where you can see `pyproject.toml`, execute the following commands in your terminal:
+```
+# Set python version
+pyenv local 3.10.6
+# Tell Poetry to use it
+poetry env use 3.10.6
+# Install all dependencies
+poetry install 
+# Activate your environment
+poetry shell
+
+```
+
+## Running the Experiments
+
+```bash
+
+#to run the experiments for the centralized model  with customized hyperparameters
+#write a command that looks like that in your terminal
+python -m hfedxgboost.main --config-name Centralized_Baseline dataset=<name of the config file that contain the dataset name, task,...> xgboost_params_centralized=<name of the config file containing the xgboot customized hyperparameters>
+#e.g
+#to run the centralized model with customized hyperparameters for cpusmall dataset, it should give 7 test MSE which is better than the 9 MSE that #the default centralized model hyperparameters give.
+python -m hfedxgboost.main --config-name Centralized_Baseline dataset=cpusmall xgboost_params_centralized=cpusmall_xgboost_centralized
+
+#to run the federated version for any dataset with no.of clients
+python -m hfedxgboost.main dataset=<name of the config file that contain the dataset name, task,...> clients=<name of the config file containing the customized hyperparameters and the no.of clients>
+#e.g
+#to run the federated version for a9a dataset with 5 clients
+python -m hfedxgboost.main dataset=a9a clients=a9a_5_clients
+
+#if you wish to change any parameters from any config file from the terminal, then you should follow this formula
+python -m hfedxgboost.main folder=config_file_name folder.parameter_name=its new value
+#e.g:
+python -m hfedxgboost.main --config-name Centralized_Baseline dataset=abalone xgboost_params_centralized=abalone_xgboost_centralized xgboost_params_centralized.max_depth=8 dataset.train_ratio=.80
+```
+
+
+## Expected Results
+
+
+```bash
+#to run all the experiments for the centralized model with the original paper config for all the datasets
+#gives the output shown in Table 1
+python -m hfedxgboost.main --config-name centralized_basline_all_datasets_paper_config
+
+#results for a9a dataset in table 2 
+python -m hfedxgboost.main --multirun clients=a9a_2_clients,a9a_5_clients,a9a_10_clients dataset=a9a
+
+#results for cod_rna dataset in table 2
+python -m hfedxgboost.main --multirun clients=cod_rna_2_clients,cod_rna_5_clients,cod_rna_10_clients dataset=cod_rna
+
+#results for ijcnn1 dataset in table 2
+python -m hfedxgboost.main --multirun clients=ijcnn1_2_clients,ijcnn1_5_clients,ijcnn1_10_clients dataset=ijcnn1
+
+#results for space_ga dataset in table 3
```

Comment:
@Aml-Hassan-Abd-El-hamid , maybe add a brief comment here indicating that the best MSE usually happens in early FL rounds (instead in the last round/s)?

Reply:
@jafermarq Thank you very much for your suggestion, I just updated the Readme with that note.